.gitignore

```diff
@@ -1,2 +1,35 @@
+# Prerequisites
 build/
 .vscode
+.DS_Store
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
+*.smod
```

Comment:
very thorough.  I would lose the fortran modules personally and I'm not sure what a few of the extensions correspond to but they do no harm what so ever.  Keep it like this if you want... you never know!

Reply:
just realised this is a merge of dev to master. whoops silly me.  Obviously I've reviewed this b4 :P

Reply:
Yeah ... Thats Cool.